kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/RemoveFile.java

```diff
@@ -15,10 +15,16 @@
  */
 package io.delta.kernel.internal.actions;
 
+import io.delta.kernel.data.MapValue;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.util.StatsUtils;
+import io.delta.kernel.internal.util.VectorUtils;
+import io.delta.kernel.statistics.DataFileStatistics;
 import io.delta.kernel.types.*;
+import java.util.Optional;
 
 /** Metadata about {@code remove} action in the Delta Log. */
-public class RemoveFile {
+public class RemoveFile extends RowBackedAction {
```

Comment:
IMO this should be a succinct PR unto itself and should be tested with unit tests

Reply:
Added the `RemoveFileSuite` in this PR as we discussed